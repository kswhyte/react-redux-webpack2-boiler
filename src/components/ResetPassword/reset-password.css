@import "imports/_helpers.css";
@import "imports/_layout.css";
@import "imports/_form-elements.css";

.reset-password-wrapper {
  & .input-row:first-child {
    text-align: center;
  }

  & .back-to-sign-in-row {
    margin-top: 46px;
  }
}
