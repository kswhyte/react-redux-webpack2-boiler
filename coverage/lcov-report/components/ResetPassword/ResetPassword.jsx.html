<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/ResetPassword/ResetPassword.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/ResetPassword</a> ResetPassword.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
&nbsp;
import HgRow from '@hg/three-ui/HgRow';
import TextInput from '@hg/three-ui/HgInputsV2/TextInput';
&nbsp;
import './reset-password.css';
&nbsp;
const ResetPassword = props =&gt; {
  return (
    &lt;div className="reset-password-wrapper"&gt;
      &lt;h2&gt;RESET YOUR PASSWORD&lt;/h2&gt;
      &lt;h3&gt;Just enter your email and we&amp;#39;ll send you instructions.&lt;/h3&gt;
&nbsp;
      &lt;form&gt;
        &lt;HgRow&gt;
          &lt;div className="col-sm-6"&gt;
            &lt;TextInput label="Your Email" placeholder="Your Email" /&gt;
          &lt;/div&gt;
        &lt;/HgRow&gt;
        {/* TODO: for RESET-btn, onClick sends a dispatch to
                  1. update client credentials
                  2. reroute to PassWordConfirm component */}
        &lt;button onClick={props.resetPassword}&gt;RESET PASSWORD&lt;/button&gt;
      &lt;/form&gt;
&nbsp;
      &lt;Link to="/login"&gt;Back to Sign in&lt;/Link&gt;
    &lt;/div&gt;
  );
};
&nbsp;
ResetPassword.propTypes = {
  resetPassword: PropTypes.func.isRequired
};
&nbsp;
export default ResetPassword;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Aug 09 2017 13:06:30 GMT-0600 (MDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
