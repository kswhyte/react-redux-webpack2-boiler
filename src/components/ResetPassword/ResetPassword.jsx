import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import isEmail from '../../helpers/isEmail';

import './reset-password.css';

const propTypes = {
  resetPassword: PropTypes.func,
  validationError: PropTypes.func,
  validationMessage: PropTypes.string,
  submitPasswordRecoveryEmail: PropTypes.func
};

const ResetPassword = props => {
  return (
    <div className="reset-password-wrapper card">
      <img src="images/pattern-dots.png" className="dots-pattern" />
      <div className="row">
        <h2>RESET YOUR PASSWORD</h2>
        <div className="page-instructions col-xs-offset-3 col-xs-6">
          <p>Just enter your email and we&#39;ll send you instructions.</p>
        </div>
      </div>

      {props.validationMessage &&
        <div className="sm-spacer validation-error">
          <h5>
            {props.validationMessage}
          </h5>
        </div>}

      <form
        onSubmit={e => {
          e.preventDefault();
          const userEmail = escape(document.querySelector('#reset-email').value);
          const emailValidation = isEmail(userEmail);
          if (!emailValidation) {
            props.validationError('The email entered is not recognized. Please enter a valid email address.');
            return false;
          }
          props.validationError('');
          const userData = {
            userEmail
          };
          props.submitPasswordRecoveryEmail(userData);
          props.resetPassword;
        }}
      >
        <div className="row lg-spacer">
          <div className="col-sm-offset-3 col-sm-6 input-row">
            <input
              id="reset-email"
              className={props.validationMessage ? 'input-error form-group' : 'form-group'}
              type="text"
              placeholder="Your Email Address"
              label="Your Email"
              required
            />
          </div>
        </div>
        <div className="row lg-spacer">
          <div className="col-xs-offset-3 col-xs-6">
            <button className="primary-btn" type="submit">
              RESET PASSWORD
            </button>
          </div>
        </div>
      </form>

      <div className="row back-to-sign-in-row">
        <Link to="/login">Back to Sign in</Link>
      </div>
    </div>
  );
};

ResetPassword.propTypes = propTypes;

export default ResetPassword;
