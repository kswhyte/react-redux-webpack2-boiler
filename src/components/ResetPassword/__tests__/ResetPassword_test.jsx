import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import ResetPassword from '../ResetPassword';

// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';
import { StaticRouter } from 'react-router';
import { shallow, mount } from 'enzyme';
import expect from 'expect';

const stories = storiesOf('ResetPassword', module);
const props = {
  resetPassword: () => {},
  validationError: () => {},
  validationMessage: 'message',
  submitPasswordRecoveryEmail: () => {}
}
stories.add('ResetPassword Page Story', () => {

  const resetPasswordStory = (
    <StaticRouter context={{}}>
      <ResetPassword {...props} />
    </StaticRouter>
  );

  specs(() =>
    describe('Show a successful alert', () => {
      it('Should render the ResetPassword component without crashing', () => {
        let output = mount(resetPasswordStory);
        expect(output.find('.reset-password-wrapper').length).toEqual(1);
      });
      it('Should have a header element that says "Reset Your Password"', () => {
        let output = mount(resetPasswordStory);
        expect(output.find('h2').text()).toContain('RESET YOUR PASSWORD');
      });
      it('Should have a <p> tag that says "Just enter your email address and we\'ll send you instructions"', () => {
        let output = mount(resetPasswordStory);
        expect(output.find('p').text()).toContain('Just enter your email and we\'ll send you instructions.');
      });
      it('Should have one input form field', () => {
        let output = mount(resetPasswordStory);
        expect(output.find('input').length).toEqual(1);
      });
      it('Should have one input form field: username/email', () => {
        let output = mount(resetPasswordStory);
        expect(output.find('#reset-email').length).toEqual(1);
      });
      it('Should have one call-to-action "Reset Password" submit button', () => {
        let output = mount(resetPasswordStory);
        expect(output.find('button').length).toEqual(1);
        expect(output.find('[type="submit"]').length).toEqual(1);
      });
      it('Should have one link: "Back to Sign in"', () => {
        let output = mount(resetPasswordStory);
        expect(output.find('a').length).toEqual(1);
        expect(output.find('a').text()).toContain('Back to Sign in');
      });
    })
  );
  return resetPasswordStory;
});
