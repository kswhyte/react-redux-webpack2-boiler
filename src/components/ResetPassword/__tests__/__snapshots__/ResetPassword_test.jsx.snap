// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Page Story 1`] = `
<div
  className="reset-password-wrapper card"
>
  <img
    className="dots-pattern"
    src="images/pattern-dots.png"
  />
  <h2>
    RESET YOUR PASSWORD
  </h2>
  <div
    className="page-instructions col-xs-offset-3 col-xs-6"
  >
    <p>
      Just enter your email and we'll send you instructions.
    </p>
  </div>
  <form>
    <div
      className="row lg-spacer"
    >
      <div
        className="col-sm-offset-3 col-sm-6 input-row"
      >
        <input
          className="form-group hg-input v2"
          id="reset-email"
          label="Your Email"
          placeholder="Your Email Address"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="row lg-spacer"
    >
      <div
        className="col-xs-offset-3 col-xs-6"
      >
        <button
          className="primary"
          onClick={undefined}
        >
          RESET PASSWORD
        </button>
      </div>
    </div>
  </form>
  <div
    className="row back-to-sign-in-row"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      Back to Sign in
    </a>
  </div>
</div>
`;
